lua/lualine/components/diagnostics.lua

```diff
@@ -0,0 +1,41 @@
+-- Copyright (c) 2020-2021 hoob3rt
+-- MIT license, see LICENSE for more details.
+
+local function diagnostics(options)
+  return function()
+    local error_count, warning_count, info_count = 0,0,0
+    if options.sources~=nil then
+      for _, source in ipairs(options.sources) do
+        if source == 'lsp' then
+          error_count = error_count +  vim.lsp.diagnostic.get_count(0, 'Error')
```

Comment:
I donn't think adding the numbers is a good idea. As multiple sources most
likely have common errors . How about displying all of them . Like if 2 sources
are configured then show like this `E: 5|4`

Reply:
Well, I don't agree with this one. If you have multiple sources of errors showing the same ones then what's the point of even having multiple sources.

Reply:
I donn't have multiple sources . But wonn't they pick up same errors?

Reply:
Well I use typescript a lot and while tsserver is nice for catching syntax errors I need eslint to catch styling errors.

Reply:
donn't eslint catch syntax errors . Pardon my ignorance I haven't used eslint.

Reply:
Well eslint is just and example, I've worked with a few linters which don't catch syntax errors and still had to use both lsp and ale. Eslint can catch whatever you configure it to catch :)

Reply:
> Eslint can catch whatever you configure it to catch

Then doesn't it make more sence to show output of separate sources separately?

Reply:
No, I prefer it this way